element_calcium_imaging/imaging_report.py

```diff
@@ -0,0 +1,418 @@
+            "showgrid": True,
+            "anchor": "free",
+            "overlaying": "y",
+            "side": "right",
+            "position": 1,
+            "titlefont": dict(color="#d62728"),
+            "tickfont": dict(color="#d62728"),
+        },
+        shapes=[
+            go.layout.Shape(
+                type="rect",
+                xref="paper",
+                yref="paper",
+                x0=0,
+                y0=0,
+                x1=1.0,
+                y1=1.0,
+                line={"width": 1, "color": "black"},
+            )
+        ],
+        legend={
+            "traceorder": "normal",
+            "yanchor": "top",
+            "y": 0.99,
+            "xanchor": "right",
+            "x": 0.99,
+        },
+        plot_bgcolor="rgba(0,0,0,0.05)",
+        modebar_remove=[
+            "zoom",
+            "resetScale",
+            "pan",
+            "select",
+            "zoomIn",
+            "zoomOut",
+            "autoScale2d",
+        ],
+    )
+
+
+@schema
+class ScanLevelReport(dj.Computed):
+    definition = """
+    -> imaging.Segmentation
+    ---
+    average_image: longblob
+    """
+
+    def make(self, key):
+        image = (imaging.MotionCorrection.Summary & key).fetch1("average_image")
+
+        cell_mask_ids, mask_xpix, mask_ypix = (
+            imaging.Segmentation.Mask * imaging.MaskClassification.MaskType & key
+        ).fetch("mask", "mask_xpix", "mask_ypix")
+
+        background_image = single_to_3channel_image(image, 1, 99)
+
+        background_image_with_cells_painted = paint_rois(
+            background_image, mask_xpix, mask_ypix
+        )
+        cells_maskid_image = make_maskid_image(
+            background_image[:, :, 0], cell_mask_ids, mask_xpix, mask_ypix
+        )
+        background_with_cells = alpha_combine_2images(
+            background_image, background_image_with_cells_painted
+        )
+
+        image_fig = go.Figure(
+            go.Image(
+                z=background_with_cells,
+                hovertemplate="x: %{x} <br>y: %{y} <br>mask_id: %{customdata}",
+                customdata=cells_maskid_image,
+            )
+        )
+
+        image_fig.update_layout(
+            title="Average Image with Cells",
+            xaxis={
+                "title": "X (px)",
+                "visible": True,
+                "showticklabels": True,
+                "showgrid": False,
+            },
+            yaxis={
+                "title": "Y (px)",
+                "visible": True,
+                "showticklabels": True,
+                "showgrid": False,
+            },
+            paper_bgcolor="rgba(0,0,0,0)",
+            plot_bgcolor="rgba(0,0,0,0)",
+        )
+
+        self.insert1({**key, "average_image": image_fig.to_json()})
+
+
+@schema
+class ActivityReport(dj.Computed):
+    definition = """
+    -> imaging.Segmentation.Mask
+    ---
+    activity_trace: longblob
+    """
+
+    def make(self, key):
```

Comment:
Saving a figure for each trace?  Is it really helpful? Wouldn't be more efficient to fetch all traces and plot them again?

Reply:
Plotly creates 1 go.Scatter object for each trace and they have to be added to the figure (see https://community.plotly.com/t/plotting-multiple-traces-at-once/27206). It's different than matplotlib.
I added two options: 1) store the go.Scatter objects in the database and fetch them for plotting (better for sciviz) 2) compute everything on the fly and generate the figure (better for codebook).